src/engine/enginebufferscalelinear.cpp

```diff
@@ -238,12 +220,11 @@ CSAMPLE* EngineBufferScaleLinear::do_scale(CSAMPLE* buf,
 
     // Simulate the loop to estimate how many frames we need
     double frames = 0;
-
     // We're calculating frames = 2 samples, so divide remaining buffer by 2;
-    for (int j = 0; j < iRateLerpLength; j += 2) {
-        frames += fabs((rate_add_diff * static_cast<float>(j)) /
-                        static_cast<float>(iRateLerpLength) + rate_add_old);
+    for (int j = 0; j < buf_size / 2; ++j) {
+        frames += (j * 2 * rate_diff / buf_size) + rate_old;
     }
+    frames = abs(frames);
```

Comment:
probably not.  we're supposed to take the floor, so there's a 50/50 chance we end up with an extra frame